sqlglot/generator.py

```diff
@@ -3315,8 +3315,26 @@ def use_sql(self, expression: exp.Use) -> str:
         return f"USE{kind}{this}"
 
     def binary(self, expression: exp.Binary, op: str) -> str:
-        op = self.maybe_comment(op, comments=expression.comments)
-        return f"{self.sql(expression, 'this')} {op} {self.sql(expression, 'expression')}"
+        sqls: t.List[str] = []
+        stack: t.List[t.Union[str, exp.Expression]] = [expression]
+        subclass = type(expression)
+        operator = expression.args.get("operator")
+
+        while stack:
+            node = stack.pop()
+
+            if isinstance(node, subclass) and (
+                (operator is None) or (operator == node.args.get("operator"))
```

Comment:
what happens when operator is None?

Reply:
If operator is None, then it only depends on the node's type. E.g. if this function was called from `sql_and`, then this code will
- directly process `And` nodes inside the loop
- call `.sql` if either operand is not an And

The `operator` check is there because `Operator` expressions produce different `op` values if they have different operators, e.g. `Operator(+)` vs `Operator(&)`.

I think this can be made a lot cleaner if we remove `op` as a parameter and instead compute it for each node within the loop, but I didn't have time to try this out this weekend.

Reply:
What is the `operator` arg referring to? I've not been able to find an `exp.Binary` expression with it defined, it should only exist in `exp.Operator` which is Postgres specific syntax afaict.

I think your intention here is to generate chains of identical binary nodes (e.g. connected concats) but `operator` is not defined so it will always be `None`

Reply:
My initial thought was to turn `op` into an property which would allow us to do `op = expression.op`,  but different dialects might use different symbols for the same node e.g. `exp.PropertyEQ` is defined differently in the Materialize dialect.

Another idea might be to create a `Generator` mapping `exp.Binary -> op (str)` which _can_ be overridden in dialect, and it will also remove the need of having separate `<bin>_sql()` functions for each expression.